### ISSUE TITLE
--date-spec=hr changes number of unique visitors

### ISSUE THREAD

**Issue by pozitron57** [allinurl/goaccess#2558]
This code gives 482 unique visitors for 13 September.

``` bash
zcat -f /var/log/nginx/access.log.1 | \
grep -v -f /home/slisakov/scripts/exclude-list.txt |\
goaccess \
  --http-protocol=no \
  --http-method=no \
  --ignore-panel=HOSTS \
  --date-spec=hr \
  --ignore-panel=REFERRING_SITES \
  --sort-panel=REQUESTS,BY_VISITORS,DESC \
  --anonymize-ip \
  --exclude-ip 176.190.94.0-176.190.94.255\
  --ignore-crawlers \
  --log-format COMBINED \
  --output /var/www/html/yesterday/index.html \
  -
```

If the `--date-spec=hr` option is omitted, number of unique visitors changes to 375, leaving other fields (total requests, valid requests etc.) intact.

What can be the reason of this difference?

(My confusion in https://github.com/allinurl/goaccess/issues/2554 actually stems from this issue)

**Comment by allinurl** [comment]
I understand. So, if you're using the `--date-spec=hr` option, the numbers will vary. Are you suggesting that you'd like the unique visitor count to be grouped by date instead of the current behavior?

**Comment by pozitron57** [comment]
> I understand. So, if you're using the `--date-spec=hr` option, the numbers will vary. Are you suggesting that you'd like the unique visitor count to be grouped by date instead of the current behavior?

Yes.

**Comment by tyctor** [comment]
Hi @allinurl 

I think that behavior for **Unique Visitors** (number in Overall Analyzed Requests panel) should be to group by same IP and agent, without date

**Comment by allinurl** [comment]
I wonder if this could be included in #2669. What are your thoughts?

**Comment by tyctor** [comment]
yes, it looks good, will be valid pattern without date? `--unique-visitors-key=IP|UA`

and also which panel(s) will be affected by value in this option? 
**Unique Visitors** (number in Overall Analyzed Requests panel) or any other?

**Comment by allinurl** [comment]
@tyctor, it should permit that. All the panes will be impacted since they rely on that key to calculate the visitors per request.

**Comment by tyctor** [comment]
ok, understand, do you plan to include this feature in next release?

**Comment by allinurl** [comment]
@tyctor, let me see if I can fit this into the next release. I've been working on some major plumbing for some exciting features. [Stay tuned](https://goaccess.io/github)!

**Comment by tyctor** [comment]
That's fine, I just asked
